Ambit/Source/Ambit/Mode/ConfigImportExport.spec.cpp

```diff
@@ -1017,4 +1017,101 @@ void ConfigImportExportSpec::Define()
             JsonContent = nullptr;
         });
     });
+
+    Describe("OnExportMap()", [this]()
+    {
+        BeforeEach([this]()
+        {
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Set AWS S3 bucket to store test data
+            AmbitMode->UISettings->AwsRegion = "us-west-2";
+            AmbitMode->UISettings->S3BucketName = "ambit-unit-test-data";
+
+            // Set target platform for export
+            AmbitMode->UISettings->ExportPlatforms.SetWindows(true);
+
+            // Generate World
+            World = FAutomationEditorCommonUtils::CreateNewMap();
+
+            // Generate Exporter
+            Exporter = NewObject<UMockableConfigImportExport>();
+        });
+
+        It("Should Return FReply::Handled", [this]()
+        {
+            const FReply Reply = Exporter->OnExportMap();
+
+            TestTrue("Reply should be true on return", Reply.IsEventHandled());
+        });
+
+        AfterEach([this]()
+        {
+            GLevelEditorModeTools().DeactivateMode(FAmbitMode::EM_AmbitModeId);
+            Exporter = nullptr;
+            JsonContent = "";
+        });
+    });
+
+    Describe("OnExportGltf()", [this]()
+    {
+        BeforeEach([this]()
+        {
+            // Generate World
+            World = FAutomationEditorCommonUtils::CreateNewMap();
+
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Set AWS S3 bucket to store test data
+            AmbitMode->UISettings->AwsRegion = "us-west-2";
+            AmbitMode->UISettings->S3BucketName = "ambit-unit-test-data";
+
+            // Generate Exporter
+            Exporter = NewObject<UMockableConfigImportExport>();
+        });
+
+        It("Should fail if no mesh to export", [this]()
+        {
+            AddExpectedError("Cannot find a static mesh to export.", EAutomationExpectedErrorFlags::Exact, 1);
+            Exporter->OnExportGltf();
+        });
+
+        It("Should succeed after test map is loaded", [this]()
+        {
+            FAutomationEditorCommonUtils::LoadMap("/Ambit/Test/Maps/ProceduralPlacementTestMap");
+
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
```

Comment:
These next few lines look like they just duplicate what was already done in the BeforeEach() method. Is there a reason these are needed here? Is there a way to refactor to avoid the repetition?

Reply:
This is done because `LoadMap()` seems to "reset" the editor so the setup for AmbitMode needs to be repeated.

Reply:
In that case, if the repetition is required I suggest encapsulating it in a helper function and calling that helper function from wherever that functionality is needed, including in your `BeforeEach()` methods. It would also be helpful to add a comment describing just what you told me about why it's needed so that others don't question it like I did.

Reply:
I couldn't find a viable way to create helper functions in UE automation tests. I have included mock S3 calls in the latest version and this has reduced a lot of repetitive code, especially wrt to AmbitMode setup, since that is no longer needed.

Only the mock S3 calls are repeated across both tests, and this is expected since they are different tests.

Reply:
BTW, I figured out how to define and use helper functions in a spec-style test. You define the function inside the `BEGIN_DEFINE_SPEC() / END_DEFINE_SPEC()` block as shown below...

```
#include "AmbitWeatherParameters.h"

#include "Misc/AutomationTest.h"

BEGIN_DEFINE_SPEC(KSchultzSpec, "KSchultz.AmbitWeatherParameters",
            EAutomationTestFlags::ProductFilter | EAutomationTestFlags::ApplicationContextMask)


FString GetGreeting()
{
    return TEXT("Hello");
}

END_DEFINE_SPEC(KSchultzSpec)

void KSchultzSpec::Define()
{
    Describe("Using a helper function", [this]()
    {
        It("works", [this]()
        {
            FString value = GetGreeting();
            TestEqual("helper return value", value, "Hello");
        });
    });
}
```

Reply:
This is incredibly helpful. I'm going to add this to our UE cookbook for other people as well.